root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -1787,3 +1776,118 @@ test('Cleanup', function (t) {
 
     t.end();
 });
+
+test('URL cleanup component: auto-select, clean-up, and validation', {}, function (t) {
+    const test_data = [
+        // AllMusic
+        {
+            i_url: 'http://www.allmusic.com/album/here-comes-the-sun-mw0002303439/releases',
+            i_ent: 'release_group',
+            match: 'allmusic',
+            clean: 'http://www.allmusic.com/album/mw0002303439',
+            valid: ['release_group']
+        },
+        // Resident Advisor (RA)
+        {
+            i_url: 'https://www.residentadvisor.net/reviews/7636',
+            i_ent: 'release_group',
+            match: 'review',
+            i_rel: 'otherdatabases',
+            valid: []
+        },
+    ];
+
+    const reltypes_by_uuid = _.reduce(LINK_TYPES, function(results, entities, name) {
+        _.each(entities, function(uuid) {
+            (results[uuid] || (results[uuid] = [])).push(name);
+        });
+        return results;
+    }, {});
+
+    var previous_match_tests = [];
+
+    function doMatchSubtest(st, ent, url, label, exp_match) {
```

Comment:
Yes, I will rename also `exp_match` to the new (still undecided) property name for `match`.